pkg/protocol/handshake/message_client_key_exchange.go

```diff
@@ -24,8 +24,21 @@ func (m MessageClientKeyExchange) Type() Type {
 // Marshal encodes the Handshake
 func (m *MessageClientKeyExchange) Marshal() ([]byte, error) {
 	switch {
-	case (m.IdentityHint != nil && m.PublicKey != nil) || (m.IdentityHint == nil && m.PublicKey == nil):
+	case (m.IdentityHint == nil && m.PublicKey == nil):
 		return nil, errInvalidClientKeyExchange
+	case (m.IdentityHint != nil && m.PublicKey != nil):
+		out := make([]byte, 2+len(m.IdentityHint)+1+len(m.PublicKey))
```

Comment:
Can you add some space to these operations? Maybe put allocations first, and then do copies? Somewhat harder to follow with no empty lines.

These are subjective, so feel free to push back.
* Use offsets instead of two slices. Users that don't know Go as well can be confused about reference vs value.
* Don't allocate `lenIdentityHint` and `lenPublicKey`. As a reader just have to go back and forth to confirm that everything is right.

Reply:
fixed